src/blockchain/log-processors/market-created.ts

```diff
@@ -141,26 +141,25 @@ export function processMarketCreatedLog(db: Knex, augur: Augur, log: FormattedEv
   });
 }
 
+async function marketCreatedLogRemoval(db: Knex, augur: Augur, log: FormattedEventLog): Promise<void> {
+  const marketId = log.market;
+  await db.from("markets").where({ marketId }).del();
+  await db.from("outcomes").where({ marketId }).del();
+  await db.from("token_supply").whereIn("token", (queryBuilder) => {
+    return queryBuilder.select("contractAddress").from("tokens").where({ marketId});
+  }).del();
+  await db.from("tokens").where({ marketId }).del();
+  await db.from("market_state").where({ marketId }).del();
+
+  const searchProvider = createSearchProvider(db);
+  if (searchProvider !== null) {
+    await searchProvider.removeSeachData(log.market);
+  }
+}
+
 export function processMarketCreatedLogRemoval(db: Knex, augur: Augur, log: FormattedEventLog, callback: ErrorCallback): void {
-  series([
-    (next: AsyncCallback): void => {
-      db.from("markets").where({ marketId: log.market }).del().asCallback(next);
-    },
-    (next: AsyncCallback): void => {
-      db.from("outcomes").where({ marketId: log.market }).del().asCallback(next);
-    },
-    (next: AsyncCallback): void => {
-      db.from("tokens").where({ marketId: log.market }).del().asCallback(next);
-    },
-    (next: AsyncCallback): void => {
-      db.from("market_state").where({ marketId: log.market }).del().asCallback(next);
-    },
-    (next: AsyncCallback): void => {
-      db.from("search_en").where({ marketId: log.market }).del().asCallback(next);
-    },
-  ], (err) => {
-    if (err) callback(err);
+  marketCreatedLogRemoval(db, augur, log).then(() => {
     augurEmitter.emit(SubscriptionEventNames.MarketCreated, log);
```

Comment:
emitting here works for market log removal, but in other places, we add data from the context of the processor, so it wouldn't be available here. i wonder if we should make this a responsibility of the promise (and then, once they are all pormises, just have the log-processor use the promise directly.

Reply:
Like we resolve the promise with the transformed data?

Reply:
yea i think in the end, we the function that calls into the blockchain processor take it as a promise directly (once they are all promises) so we don't have to have a shim everywhere

Reply:
I like that.

Reply:
Random thought -- can we use Typescript generics to kinda give us the ability to export either a function-with-callback or a async-function??

Reply:
Yes, but it will still come down to:
```
if isCallbackStyle(processor) processor(...)
else isPromiseStyle(processor) processor(...)